tests/integration/cqlengine/columns/test_validation.py

```diff
@@ -337,50 +338,235 @@ def test_default_zero_fields_validate(self):
         it.validate()
 
 
-class TestText(BaseCassEngTestCase):
+class TestAscii(BaseCassEngTestCase):
 
     def test_min_length(self):
-        # not required defaults to 0
-        col = Text()
-        col.validate('')
-        col.validate('b')
+        """ Test arbitrary minimal lengths requirements. """
+        Ascii(min_length=0).validate('')
+        Ascii(min_length=0).validate(None)
+        Ascii(min_length=0).validate('kevin')
+
+        Ascii(min_length=1).validate('k')
+
+        Ascii(min_length=5).validate('kevin')
+        Ascii(min_length=5).validate('kevintastic')
 
-        # required defaults to 1
         with self.assertRaises(ValidationError):
-            Text(required=True).validate('')
+            Ascii(min_length=1).validate('')
+
+        with self.assertRaises(ValidationError):
+            Ascii(min_length=1).validate(None)
+
+        with self.assertRaises(ValidationError):
+            Ascii(min_length=6).validate('')
+
+        with self.assertRaises(ValidationError):
+            Ascii(min_length=6).validate(None)
+
+        with self.assertRaises(ValidationError):
+            Ascii(min_length=6).validate('kevin')
+
+        with self.assertRaises(ValueError):
+            Ascii(min_length=-1)
+
+    def test_max_length(self):
+        """ Test arbitrary maximal lengths requirements. """
+        Ascii(max_length=0).validate('')
+        Ascii(max_length=0).validate(None)
+
+        Ascii(max_length=1).validate('')
+        Ascii(max_length=1).validate(None)
+        Ascii(max_length=1).validate('b')
+
+        Ascii(max_length=5).validate('')
+        Ascii(max_length=5).validate(None)
+        Ascii(max_length=5).validate('b')
+        Ascii(max_length=5).validate('blake')
+
+        with self.assertRaises(ValidationError):
+            Ascii(max_length=0).validate('b')
+
+        with self.assertRaises(ValidationError):
+            Ascii(max_length=5).validate('blaketastic')
+
+        with self.assertRaises(ValueError):
+            Ascii(max_length=-1)
+
+    def test_length_range(self):
+        Ascii(min_length=0, max_length=0)
+        Ascii(min_length=0, max_length=1)
+        Ascii(min_length=10, max_length=10)
+        Ascii(min_length=10, max_length=11)
+
+        with self.assertRaises(ValueError):
+            Ascii(min_length=10, max_length=9)
+
+        with self.assertRaises(ValueError):
+            Ascii(min_length=1, max_length=0)
+
+    def test_type_checking(self):
+        Ascii().validate('string')
+        Ascii().validate(u'unicode')
+        Ascii().validate(bytearray('bytearray', encoding='ascii'))
+
+        with self.assertRaises(ValidationError):
+            Ascii().validate(5)
+
+        with self.assertRaises(ValidationError):
+            Ascii().validate(True)
+
+    def test_unaltering_validation(self):
+        """ Test the validation step doesn't re-interpret values. """
+        self.assertEquals(Ascii().validate(''), '')
```

Comment:
assertEquals needs to be swapped over to assertEqual, as assertEquals is slated for depreciation in the future.